File: ssluzba konacno/src/test/javascript/e2e/pages/globalzavrsni/menu.page.js
var MenuPage = function() {};
var utils = require('../utils.js');

MenuPage.prototype = Object.create({}, {
    
 
		

ikonicaZaProsirenje: {
        get: function() {
            return utils.waitToBeClickable(by.css("a.sidebar-toggle"), 10000);
        }
    },


languages: {
        get: function() {
            return utils.waitToBeClickable(by.css("button.btn-default.dropdown-toggle"));
        }
    },
	
	
srpskiLatinica: {
        get: function() {
            return utils.waitToBeClickable(by.xpath("//a [@translate='BUTTON_LANG_SR_LAT']"), 20000);
        }
    },
	
	
srpskiCirilica: {
        get: function() {
            return utils.waitToBeClickable(by.xpath("//a [@translate='BUTTON_LANG_SR_CYR']"), 10000);
        }
    },
	
	
engleski: {
        get: function() {
            return utils.waitToBeClickable(by.xpath("//a [@translate='BUTTON_LANG_EN']"), 10000);
        }
    },



dropdownmenu: {
        get: function() {
            return utils.waitToBeClickable(by.xpath("//*[@id='main-page']/header/nav/div/ul[2]/li"), 10000);
        }
    },
	

image: {
        get: function() {
            return utils.waitForElementPresence(by.css("img-circle"), 10000);
        }
    },


 userManual: {
        get: function() {
            return utils.waitForElementPresence(by.xpath("//span[@translate='USER_MANUAL']"), 10000);
        }
    },
	

 contact: {
        get: function() {
            return utils.waitToBeClickable(by.xpath("//span[@translate='USER_SUPPORT']"), 10000);
        }
    },

 changePassword: {
        get: function() {
            return utils.waitToBeClickable(by.xpath("//span[@translate='CHANGE_PASSWORD']"), 10000);
        }
    },
    
logoutBtn: {
        get: function() {
            return utils.waitForElementPresence(by.css("span[translate='LOGOUT']"), 10000);
        }
    },
   


institucijaBtn: {
        get: function() {
            return utils.waitToBeClickable(by.xpath("//a [@href='#/admin-institution/']"), 10000);
        }
    },

istrazivaciBtn: {
        get: function() {
            return utils.waitToBeClickable(by.xpath("//a [@href='#/persons']"), 10000);
        }
    }

	
	
});

// Export klase
module.exports = MenuPage;

File: ssluzba konacno/src/main/java/global/MenuPage.java
package global;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

import helpers.Utils;

public class MenuPage {
	private WebDriver driver;

	public MenuPage(WebDriver driver) {
		super();
		this.driver = driver;
	}

	
	public WebElement getMenyIcon() {
		return Utils.waitToBeClickable(driver, By.xpath("//span [contains(text(), 'Toggle navigation')]"), 10);
	}
	
	
	public WebElement getLanguages() {
		return Utils.waitToBeClickable(driver, By.xpath("//*[@id='main-page']/header/nav/div/ul[1]/div/button/span[1]/img"), 10);
	}
	public WebElement getLanguagesSerbLat() {
		return Utils.waitToBeClickable(driver, By.xpath("//a [@translate='BUTTON_LANG_SR_LAT']"), 10);
	}
	public WebElement getLanguagesSerbCir() {
		return Utils.waitToBeClickable(driver, By.xpath("//a [@translate='BUTTON_LANG_SR_CYR']"), 10);
	}
	public WebElement getLanguagesEng() {
		return Utils.waitToBeClickable(driver, By.xpath("//a [@translate='BUTTON_LANG_EN']"), 10);
	}

	
	public WebElement getDropdown() {
		return Utils.waitToBeClickable(driver, By.xpath("//*[@id='main-page']/header/nav/div/ul[2]/li"), 10);
	}
	
	
	
	
	

	public WebElement getLogout() {
		return Utils.waitToBeClickable(driver, By.xpath("//span [@translate='LOGOUT']"), 10);
	}

	

}
